### PR TITLE
Replace IBAction with RxCocoa 

### PR DESCRIPTION
**Pull request by davidlamys**
#25 :)

### CODE REVIEW

**Comment by davidlamys** [comment]
I think I closed my PR when I force pushed. >.<

File: ChessClock/Clocks/ViewController.swift [hunk 1]
```diff
@@ -26,24 +27,30 @@ class ViewController: UIViewController {
...
+            .map({ _ in ClocksEvents.pause })
+            
+        let restart = resetButton.rx.tap
+            .map({ _ in ClocksEvents.restart })
+            
+        let resume = resumeButton.rx.tap
+            .map({ _ in ClocksEvents.resume })
+            
+        Observable.merge(startTop, startBottom, pause, restart, resume)
+            .bind(to: presenter.clocksStateBehaviourSubject)
```
Comment (joanb) [inline ChessClock/Clocks/ViewController.swift:53]: Really nice!!!! We finally deal with streams starting from user interactions 🎉 🎉 
Thank you for your contribution, as always 💪

**Comment by davidlamys** [comment]
Dammit the whitespace is killing me. hahah.. i'm way too used to my git hooks in my normal projects.

**Comment by joanb** [comment]
Hahaha nice! 
Maybe we can add a symlink workaround in the future for adding static code analysis with git hooks  😄

**Comment by davidlamys** [comment]
> Hahaha nice!
> Maybe we can add a symlink workaround in the future for adding static code analysis with git hooks 😄

oh can you elaborate? I haven't heard of this before. May be a link to some article? :)

**Comment by joanb** [comment]
Sure, here you have[ one link](https://rock-it.pl/automatic-code-quality-checks-with-git-hooks/) I followed the first time I used this.
It's not that magic, and can come handy in some situations, although it can be a bad idea if used wrong :)
Have fun with this! 
Of course if you have any question don't hesitate to ask!! 😄

**Comment by davidlamys** [comment]
Oh I implemented a variant of that at work. But whoever checks out would
still have to move the script, authorize it and stuff. I thought you meant
we could validate it on the server side while hosting on github...and
perhaps even automatically suggests a commit. that would be way cool..

Can you elaborate the part about creating a symlink workaround?

**Comment by joanb** [comment]
> I thought you meant
we could validate it on the server side while hosting on github...and
perhaps even automatically suggests a commit. that would be way cool..

Yes... that would be awesome...

> Can you elaborate the part about creating a symlink workaround?

The symlink thing is the one explained in the article, just a project "setup" everyone has to do locally in order to activate the hooks... I thought you were referring to something like this when you said:

> i'm way too used to my git hooks in my normal projects

-----------------------
Anyway there should be something out there that does this kind of clean ups automatically 🤔 I'll take a look when I have the time :)

**Pull request merged**